### DISCUSSION TITLE
Why not keep Radix syntax of importing components?

### DISCUSSION THREAD

**Discussion by skriken** [tremorlabs/tremor-npm#1119] [General]
Hi team.

I hope this has not been asked previously, and if so, feel free to link me to it.
I was wondering why Tremor doesn't utilize the same conventions as Radix in terms of syntax of imports/exports.

To illustrate what I mean, see the code sections below.

**What we currently have:**
```tsx
import { DropdownMenu, DropdownMenuTrigger, DropdownMenuLabel, DropdownMenuContent } from "@...."

<DropdownMenu>
  <DropdownMenuTrigger></DropdownMenuTrigger>
  <DropdownMenuContent></DropdownMenuContent>
</DropdownMenu>
```

**What I want/prefer:**
```tsx
import { DropdownMenu } from "@..."

<DropdownMenu.Root>
  <DropdownMenu.Trigger></DropdownMenu.Trigger>
  <DropdownMenu.Content></DropdownMenu.Content>
</Dropdown.Menu>
```

It may not seem obvious at first, but here are some points on why I prefer this syntax over the current one:

**Readability & efficiency**
- It improves readability and my efficiency writing custom code with Tremor as everything related to a Tremor component can be used with a simple "." while coding, instead of guessing or looking up the component or searching in the docs.
- It reduces mental overhead. I no longer need to read the entire component name in the code, while I'm in the "DropdownMenu context", I can just glance at the post-dot naming and understand.

**Cleaner code - less imports**
From
![image](https://github.com/user-attachments/assets/d0113f02-f77f-412c-a10e-acd6b9bea735)
to
![image](https://github.com/user-attachments/assets/93090876-138a-4327-9d25-c8519c8bf203)

---

Hoping to learn more about this!

**Comment by severinlandolt** [comment]
Hi! This is just some leftover from the early days 2022. Might change in v4 :)
